lib/site_prism/page.rb

```diff
@@ -64,12 +64,8 @@ def load(expansion_or_html = {}, &block)
     end
 
     def displayed?(*args)
-      expected_mappings = args.last.is_a?(::Hash) ? args.pop : {}
-      seconds = !args.empty? ? args.first : Capybara.default_max_wait_time
-      raise SitePrism::NoUrlMatcherForPageError unless url_matcher
-
       begin
-        Waiter.wait_until_true(seconds) { url_matches?(expected_mappings) }
+        Waiter.wait_until_displayed?(*args, url_matcher) { url_matches? } 
```

Comment:
lets make this method operate like a predicate, drop all waiting logic

Reply:
@luke-hill I've pushed a new commit, I think it's a bit ugly but all the current tests are now passing which is a small comfort